pandas/_libs/src/parser/tokenizer.c

```diff
@@ -424,13 +424,13 @@ static void append_warning(parser_t *self, const char *msg) {
 
     if (self->warn_msg == NULL) {
         self->warn_msg = (char *)malloc(length + 1);
-        strncpy(self->warn_msg, msg, strlen(msg) + 1);
+        snprintf(self->warn_msg, length + 1, "%s", msg);
```

Comment:
1. https://stackoverflow.com/a/56782476
2. cpplint complains about using bare strcpy. 

snprintf is fine with both.